main.nf

```diff
@@ -175,41 +208,42 @@ process star {
     cpus 8
     memory '64 GB'
     time '5h'
-
+    
+    publishDir "$results_path/STAR"
+    
     input:
     file index
     file gtf
-    file trimmed_read1 from trimmed_read1
-    file trimmed_read2 from trimmed_read2
-
+    file (reads:'*') from trimmed_reads
+    set val(prefix) from name_for_star 
     output:
-    file '*.Aligned.sortedByCoord.out.bam' into bam4, bam5, bam6, bam7, bam8, bam9
-    file '*.Log.final.out' into results
-    file '*.Log.out' into results
-    file '*.Log.progress.out' into results
-    file '*.SJ.out.tab' into results
-
+    file '*.bam' into bam4, bam5, bam6, bam7, bam8, bam9
```

Comment:
can we rename these `bam4` into `bam_rseqc` etc, named after processes instead please?